src/main/java/com/rolfje/anonimatron/anonymizer/ParameterizedAnonymizer.java

```diff
@@ -0,0 +1,34 @@
+package com.rolfje.anonimatron.anonymizer;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.rolfje.anonimatron.synonyms.Synonym;
+
+/**
+ * Provides functionality for consistently anonymizing a piece of data with provided parameters.
+ * @see com.rolfje.anonimatron.anonymizer.Anonymizer
+ */
+public interface ParameterizedAnonymizer extends Anonymizer {
```

Comment:
This class may not be needed:
If we add a default method to Anonymizer with parameters, and call the old method from that, the old Anonymizers would still compile if I understand correctly. Another benefit is that we never have to check "if instanceof".

Reply:
Absolutely. A bit overengineered here ^^'